### PR TITLE
chore(telemetry): Change logging level to `DEBUG` for most messages

### LINKED ISSUE #819: `kedro-telemetry`: Revisit logging messages and make them `DEBUG`
## Description
Most logging messages in kedro-telemetry shouldn't be surfaced to the users. On top of that, there's currently a mix of logging levels in telemetry message (see context below). 

## Context
```
$ uv run kedro registry list
[08/23/24 18:13:54] INFO     Using                                                                        __init__.py:249
                             '/private/tmp/kedro-telemetry-test-starter/.venv/lib/python3.12/site-package                
                             s/kedro/framework/project/rich_logging.yml' as logging configuration.                       
[08/23/24 18:13:56] ERROR    Failed to retrieve UUID: badly formed hexadecimal UUID string                   plugin.py:81
- __default__

                    INFO     Kedro is sending anonymous usage data with the sole purpose of improving the   plugin.py:233
                             product. No personal data or IP addresses are stored on our side. If you want               
                             to opt out, set the `KEDRO_DISABLE_TELEMETRY` or `DO_NOT_TRACK` environment                 
                             variables, or create a `.telemetry` file in the current working directory with              
                             the contents `consent: false`. Read more at                                                 
                             https://docs.kedro.org/en/stable/configuration/telemetry.html                               
[08/23/24 18:13:57] WARNING  Failed to send data to Heap. Response code returned: 400, Response reason: Bad plugin.py:336
                             Request
```

## Implementation
Update the logging messages to make them `DEBUG` level, only keeping them at higher level if really necessary and the end user can actually do something about it. 
Also mention kedro and/or kedro telemetry in the message so it's clearer they come from `kedro-telemetry`.

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description

Resolve #819 

## Development notes

Made most logging messages `DEBUG` as users can't do much about them.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by merelcht — APPROVED** [review]
LGTM! 👍

File: kedro-telemetry/tests/test_plugin.py [hunk 1]
```diff
@@ -365,11 +363,6 @@ def test_before_command_run_heap_call_error(self, mocker, fake_metadata, caplog)
 
         telemetry_hook.before_command_run(fake_metadata, command_args)
         telemetry_hook.after_command_run()
-        msg = (
```
Comment (DimedS) [inline kedro-telemetry/tests/test_plugin.py:368]: Thanks, @ankatiyar! However, I'm curious—why did we remove the message checks in the tests?

Comment (ankatiyar) [reply kedro-telemetry/tests/test_plugin.py:368]: They don't show up on the logs if the messages are at debug level

Comment (DimedS) [reply kedro-telemetry/tests/test_plugin.py:368]: Is it possible to change the `caplog` level as done in the previous tests, using
`with caplog.at_level(logging.INFO):`

Comment (ankatiyar) [reply kedro-telemetry/tests/test_plugin.py]: Changed the logging level for tests instead - I was having some trouble with changing it at individual levels. Does this work?

Comment (DimedS) [reply kedro-telemetry/tests/test_plugin.py]: Thanks! I don't see any issues with that. I believe it's a good solution!

**Review by DimedS — APPROVED** [review]
Thanks, @ankatiyar ! great job!

**Pull request merged**